### ISSUE TITLE
Scenarios: Use data previously created in the scenario as input to other scenario data

### ISSUE THREAD

**Issue by KrisCoulson** [redwoodjs/redwood#2422]
Currently trying to use scenarios for testing nested data is a bit lack luster. When working with some relationships we need the item created to be created with addition relationships. It would be awesome if we were able to create scenario data and then use it within the scenario.

@cannikin had some initial thoughts which can be found here on the [discord channel](https://discord.com/channels/679514959968993311/716252919875240007/827680139268849675)
![image](https://user-images.githubusercontent.com/6943688/116765269-2ba28280-a9d9-11eb-9e34-756c74a737fe.png)

Additionally it might be nice to have someway to access the context.currentUser in the scenario.

**Comment by KrisCoulson** [comment]
Just circling back on this because I was working on a new feature and came across a commented-out scenario block of code that was linked to this. An additional use case for this is for multi-tenancy applications. I currently have an `organizationId` that is required when inserting rows into the database.  The current problem is I need to create the organization before I can connect the record.

It would be really nice if there was a way to create dependant data before the scenario is executed.

I find myself always reaching for scenarios but unable to use them

Maybe the ability to pass a function or and obj to defineScenario

```
 export const standard = defineScenario(async (db) => {
    const org = await db.create.org()
 
     return {
         event: {
             one: {
                 orgId: org.id
                 name 'Test Event'
             }
         },
         team: {
             one: {
                 orgId: org.id,
                 name: 'Test Team
             }
         }
     }
 })
```

**Comment by cannikin** [comment]
You can create other records along with the ones you really care about using the default Prisma syntax, does that help in your situation? For example:

```javascript
export const standard = defineScenario(async (db) => {
  return {
    event: {
      one: {
        name 'Test Event',
        org: {
          create: {
            name: 'Test Org'
          }
        },
        team: {
          create: {
            name: 'Test Team'
          }
        }
      }
    }
  }
})
```

In your specific example you're using the same org in both models, which I don't know if you can do with this syntax, though!

**Comment by KrisCoulson** [comment]
@cannikin yeah that is the issue, you can't reuse the org so If you wanted to create two events in the `scenario.event.one` and `scenario.event.two` they can't be linked to the same organization

**Comment by KrisCoulson** [comment]
@cannikin I made a draft PR that I think takes a stab at this was super straight forward and aligns with what we had discussed way back when in discord. #3386

**Comment by thedavidprice** [comment]
Closed with #3386
